frontend/src/components/settings/index.tsx

```diff
@@ -133,8 +133,14 @@ export const Settings = () => {
   useEffect(() => {
     const subscription = watch((values, { name }) => {
       const [group, label] = (name as string).split(".");
-
-      debouncedUpdate(group, label, values[group][label]);
+      const currentSettingValue = values[group][label] as any;
+      // if setting value is Number but typed string convert it to number
+      const settingValue =
+        typeof values[group][label] === "string" && isNaN(currentSettingValue)
+          ? values[group][label]
+          : Number(values[group][label]);
```

Comment:
```suggestion
       const setting = settings.find(
        (setting) => setting.group === group && setting.label === label,
      )?.value;
      // if setting value is Number but typed string convert it to number
      const settingValue =
        typeof setting === "number"
          ? Number(values[group][label])
          : values[group][label];
```
Hi @abdou6666, do you think maybe that we can do something like this to resolve the data save issue